File: go/house/house.go
package house

const testVersion = 1

func Verse(line int) string {
  lyrics := map[string]string {
    "ate": "the malt",
    "killed": "the rat",
    "worried": "the cat",
    "tossed": "the dog",
    "milked": "the cow with the crumpled horn",
    "kissed": "the maiden all forlorn",
    "married": "the man all tattered and torn",
    "woke": "the priest all shaven and shorn",
    "kept": "the rooster that crowed in the morn",
    "belonged": "to the farmer sowing his corn" }

  if line <= 1 {
    return "This is the house that Jack built."
  }
  Verse(line - 1)
}

func Song() string {
  return Verse(1)
}

File: python/house/house.py
def recite(start_verse, end_verse):
    return [recite_verse(verse) for verse in range(start_verse, end_verse + 1)]

def recite_verse(verse):
    m = [('the house that Jack built', 'lay in'),
         ('the malt','ate'),
         ('the rat','killed'),
         ('the cat','worried'),
         ('the dog','tossed'),
         ('the cow with the crumpled horn','milked'),
         ('the maiden all forlorn','kissed'),
         ('the man all tattered and torn','married'),
         ('the priest all shaven and shorn','woke'),
         ('the rooster that crowed in the morn','kept'),
         ('the farmer sowing his corn','belonged to'),
         ('the horse and the hound and the horn','')]
    return " ".join([f"This is {m[s][0]}" if s == verse - 1 else f"that {m[s][1]} {m[s][0]}" for s in range(0, verse)][::-1]) + '.'
